docs/developer-docs/latest/developer-resources/error-handling.md

```diff
@@ -109,12 +118,338 @@ export default {
+
+::: note
+See the [default error classes](#default-error-classes) section for more information on the error classes provided by Strapi.
+:::
+
+::: details Example: Throwing an error in a service
+This example shows wrapping a [core service](/developer-docs/latest/development/backend-customization/services.md#extending-core-services) and doing a custom validation on the `create` method.
+
+<code-group>
+<code-block title="JAVASCRIPT">
+
+```js
+// path: ./src/api/restaurant/services/restaurant.js
+
+const utils = require('@strapi/utils');
+const { ApplicationError } = utils.errors;
+const { createCoreService } = require('@strapi/strapi').factories;
+
+module.exports = createCoreService('api::restaurant.restaurant', ({ strapi }) =>  ({
+  async create(params) {
+    let okay = false;
+
+    // Throwing an error will prevent the restaurant from being created
+    if (!okay) {
+      throw new ApplicationError('Something went wrong', { foo: 'bar' });
+    }
+  
+    const result = await super.create(params);
+
+    return result;
+  }
+});
+
+```
+
+</code-block>
+
+<code-block title="TYPESCRIPT">
+
+```js
+// path: ./src/api/[api-name]/policies/my-policy.ts
+
+import utils from '@strapi/utils';
+import { factories } from '@strapi/strapi';
+
+const { ApplicationError } = utils.errors;
+
+export default factories.createCoreService('api::restaurant.restaurant', ({ strapi }) =>  ({
+  async create(params) {
+    let okay = false;
+
+    // Throwing an error will prevent the restaurant from being created
+    if (!okay) {
+      throw new ApplicationError('Something went wrong', { foo: 'bar' });
+    }
+  
+    const result = await super.create(params);
+
+    return result;
+  }
+}));
+
 ```
 
 </code-block>
 </code-group>
 
+:::
+
+::: details Example: Throwing an error in a model lifecycle
+This example shows building a [custom model lifecyle](/developer-docs/latest/development/backend-customization/models.md#lifecycle-hooks) and being able to throw an error that stops the request and will return proper error messages to the admin panel. Generally you should only throw an error in `beforeX` lifecycles, not `afterX` lifecycles.
+
+<code-group>
+<code-block title="JAVASCRIPT">
+
+```js
+// path: ./src/api/[api-name]/content-types/[api-name]/lifecycles.js
+
+const utils = require('@strapi/utils');
+const { ApplicationError } = utils.errors;
+
+module.exports = {
+  beforeCreate(event) {
+    let okay = false;
+
+    // Throwing an error will prevent the entity from being created
+    if (!okay) {
+      throw new ApplicationError('Something went wrong', { foo: 'bar' });
+    }
+  },
+};
+
+```
+
+</code-block>
+
+<code-block title="TYPESCRIPT">
+
+```js
+// path: ./src/api/[api-name]/content-types/[api-name]/lifecycles.ts
+
+import utils from '@strapi/utils';
+const { ApplicationError } = utils.errors;
+
+export default {
+  beforeCreate(event) {
+    beforeCreate(event) {
```

Comment:
Is it correct to stack 2 beforeCreates for TypeScript?

Reply:
No that's a typo lol